pkg/snapstore/s3_snapstore.go

```diff
@@ -519,3 +595,23 @@ func isAWSConfigEmpty(config *awsCredentials) error {
 	}
 	return fmt.Errorf("aws s3 credentials: region, secretAccessKey or accessKeyID is missing")
 }
+
+// Creates SSE Credentials that are included in the S3 API calls for customer managed SSE
+// Key rotation is not handled by etcd-backup-restore, use SSE through customer managed keys at your discretion
+// See: https://github.com/gardener/etcd-backup-restore/pull/719#issuecomment-2016366462
+func getSSECreds(sseCustomerKey, sseCustomerAlgorithm *string) (SSECredentials, error) {
+	if sseCustomerKey != nil && sseCustomerAlgorithm != nil {
+		SSECustomerKeyMD5Bytes := md5.Sum([]byte(*sseCustomerKey))
+		sseCustomerKeyMD5 := base64.StdEncoding.EncodeToString(SSECustomerKeyMD5Bytes[:])
+		return SSECredentials{
+			sseCustomerKey:       *sseCustomerKey,
+			sseCustomerKeyMD5:    sseCustomerKeyMD5,
+			sseCustomerAlgorithm: *sseCustomerAlgorithm,
+		}, nil
+	}
+	if sseCustomerKey != nil || sseCustomerAlgorithm != nil {
+		return SSECredentials{}, fmt.Errorf("both sseCustomerKey and sseCustomerAlgorithm are to be provided if customer managed SSE keys are to be used")
+	}
+	// SSE Customer Managed keys not enabled
+	return SSECredentials{}, nil
```

Comment:
Early return for no SSE-C credentials can be used, instead of having it as the last condition.